### ISSUE TITLE
Provide users with a way to reset their password

### ISSUE THREAD

**Issue by yunohost-bot** [YunoHost/issues#878]
###### Original Redmine Issue: [878](https://dev.yunohost.org/issues/878)

Author Name: **alexAubin**

---

Continuation of https://github.com/YunoHost/SSOwat/issues/6

>We need to have a way for users to reset/retreive their password.
>Easyiest way to do so: send a mail to the domain admin.
>Store a fallback mail address, in th case that admin couldn't contact the user IRL.

**Comment by gnouts** [comment]
What's the status of this issue ? Is it hard to implement, from a security perspective ?

**Comment by Psycojoker** [comment]
No one has been working on yet.

Technically it's no extremely hard since it has been done a lot of time already but we have YunoHost specific questions for that (like if you lost you sso password ... that also happen to be your email password... sending you en email is a bit pointless etc...)

**Comment by gnouts** [comment]
I understand.
I was thinking of something that let users enable this feature. Enabling it would require the user to provide a recovery email which is not the yunohost one. It's something most users are used to, IMO. 

(I'm surprised this issue is not requested more ^^)

**Comment by Psycojoker** [comment]
> (I'm surprised this issue is not requested more ^^)

People tend to use YunoHost for a small group of user or only themselves in general, it's quite new that we have bigger scale usage '-'

**Comment by alexAubin** [comment]
Note that the issue is imho not trivial to solve : many people don't have a working email stack (either because they don't care so much about it, because they didn't set up the DNS records properly, or because their ISP doesn't allow to open port 25, or ...)

So sending an email to an external adress is not guaranteed to work ...

One can imagine fallback solutions such as having some sort of custom relay hosted on yunohost's  project infrastructure (with various constraints to limit abuses). Or just send an email to the admin of the instance to get the password resetted (assuming the admin reads emails). Or yunohost could simply disable the recovery password thing automatically if it's able to diagnose that the mail stack ain't properly working / configured.

**Comment by alexAubin** [comment]
(Supposedly ljf is working on this)

**Comment by zamentur** [comment]
Exact, i am currently working on finding a way to specify a specific mail for recovery in ldap.

**Comment by artybdrlt** [comment]
Hello @zamentur!
Have you found something yet regarding this issue?

**Comment by alexAubin** [comment]
Initial exploration was done in https://github.com/YunoHost/yunohost/pull/1322 and the person working on it should have a look before jumping into work

In particular, the `_smtp_is_secured_enough(..)` function at the very end of the diff

**Comment by isAAAc** [comment]
new request today in support room: https://matrix.to/#/!gvRsI7cG1o9SVr56:pijean.ovh/$O5gjBl296z6iPbv5f-0mY6Q3JoQ5h16SfLnBv5vbfiM?via=matrix.org&via=aria-net.org&via=tchncs.de
